types/hooks.d.ts

```diff
@@ -0,0 +1,235 @@
+/// <reference types="./react-spring"/>
+
+declare module 'react-spring/hooks' {
+  import {
+    SpringConfig,
+    SpringBaseProps,
+    TransitionKeyProps,
+  } from 'react-spring'
+
+  /** List from `function getForwardProps` in `src/shared/helpers` */
+  type ExcludedProps =
+    | 'to'
+    | 'from'
+    | 'config'
+    | 'native'
+    | 'onStart'
+    | 'onRest'
+    | 'onFrame'
+    | 'children'
+    | 'reset'
+    | 'reverse'
+    | 'force'
+    | 'immediate'
+    | 'impl'
+    | 'inject'
+    | 'delay'
+    | 'attach'
+    | 'destroyed'
+    | 'track'
+    | 'interpolateTo'
+    | 'autoStart'
+    | 'ref'
+  type ForwardedProps<T> = Pick<T, Exclude<keyof T, ExcludedProps>>
+  // NOTE: because of the Partial, this makes a weak type, which can have excess props
+  type InferFrom<T extends object> = T extends { to: infer TTo }
+    ? Partial<TTo>
+    : Partial<ForwardedProps<T>>
+
+  // This is similar to "Omit<A, keyof B> & B",
+  //  but with a delayed evaluation that still allows A to be inferrable
+  type Merge<A, B> = { [K in keyof A]: K extends keyof B ? B[K] : A[K] } & B
+
+  type SetUpdateFn<DS extends object> = (ds: DS) => void
+
+  // The hooks do emulate React's 'ref' by accepting { ref?: React.RefObject<Controller> } and
+  // updating it. However, there are no types for Controller, and I assume it is intentionally so.
```

Comment:
Actually, I can't find the code that was doing it anymore? 🤔

Reply:
I could implement this with a fake nominal type if you still want to keep `Controller` opaque 🤔